### PR TITLE
Porting SupportMaterial

### PR DESCRIPTION
**Pull request by Samir55**
## What's left
- [ ] Fix failing tests.

### CODE REVIEW

**Comment by Samir55** [comment]
yes, @lordofhyphens :) I was about to change that.

**Comment by lordofhyphens** [comment]
Also, `interface` is apparently a keyword in Windows (defined in `<windows.h>`. Found that out when I did a test merge from cppgui to master.

I've renamed the usages in SupportMaterial to `_interface`, but if you can think of a better renaming we'll go with that.

**Comment by lordofhyphens** [comment]
As soon as I have a working build on Windows I'll give you a commit ID to cherry pick from master.

**Comment by lordofhyphens** [comment]
4451fc424c0eef293806acb7944383629cd878a9 is the commit on slic3r/slic3r

File: src/test/libslic3r/test_support_material.cpp [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-#include <libslic3r/GCodeReader.hpp>
+#include "GCodeReader.hpp"
+#include "test_data.hpp"
 
-#include "libslic3r.h"
-#include "TriangleMesh.hpp"
-#include "Model.hpp"
-#include "SupportMaterial.hpp"
+// For IDE suggestions
+#include "../../../build/external/Catch/include/catch.hpp"
```
Comment (lordofhyphens) [inline src/test/libslic3r/test_support_material.cpp]: You probably should just not stage these lines in the future. 

This line will crash if someone builds out of tree.

**Comment by lordofhyphens** [comment]
Support is not implemented with regions so far as I am aware. If they were,
we could customize it or turn it off from modifier meshes :D

Implementing it would be lovely, but out of scope for your current work.

**Pull request closed without merging**